month/months.js

```diff
@@ -1,70 +1,93 @@
-"use strict"
+import { isSameDate, setSelectedDate, selectedDate, WEEKDAYS } from "../js/dateManipulation.js"
+import { switchWeekView } from "../js/weekView.js"
+
 
 const CALENDAR_ROWS = 6
-const DAYS_IN_WEEK = 7
 
 const today = new Date(Date.now())
-let currentMonth = today.getMonth()
-let currentYear = today.getFullYear()
+let selectedDate_month = selectedDate
 
 document.querySelector("#months-next").addEventListener('click', () => {
-    incrementMonth(1)
-    fillMonth(currentYear, currentMonth)
+    const nextDate = new Date(selectedDate_month.getFullYear(), selectedDate_month.getMonth() + 1, 1)
+    switchMonth( nextDate )
+    selectedDate_month = nextDate
+    updateLabel(selectedDate_month)
 })
 document.querySelector("#months-prev").addEventListener('click', () => {
-    incrementMonth(-1)
-    fillMonth(currentYear, currentMonth)
+    const nextDate = new Date(selectedDate_month.getFullYear(), selectedDate_month.getMonth() - 1, 1)
+    switchMonth( nextDate )
+    selectedDate_month = nextDate
+    updateLabel(selectedDate_month)
 })
 
-function updateLabel(){
+function updateLabel( date ){    
     const monthViewLabelMonth = document.querySelector("#monthView-label-month")
     const monthViewLabelYear = document.querySelector("#monthView-label-year")
-    monthViewLabelMonth.innerHTML = new Date(currentYear, currentMonth).toLocaleDateString('us-US', {month: 'long'})
-    monthViewLabelYear.innerHTML = currentYear
+    monthViewLabelMonth.innerHTML = date.toLocaleDateString('us-US', {month: 'long'})
+    monthViewLabelYear.innerHTML = date.getFullYear()
 }
 
-function incrementMonth(value){
-    currentMonth += value
-    if(currentMonth > 11){
-        currentMonth = currentMonth%12
-        currentYear++
-    }
-    if(currentMonth < 0){
-        currentMonth += 12
-        currentYear--
-    }
+export function toggleSelectedDate(element){
+    const buttons = document.querySelectorAll('#month button')
+    buttons.forEach(elem => {
+        elem.classList.remove('button_selectedDay')
+    })
+    element.classList.add('button_selectedDay')
 }
 
-function generateDayCell(content){
-    const element = 
-    `<div class="container monthView-cell">
-        <button 
-            class="button button-round monthView-button"
-            data-currentmonth=${content.currentMonth}
-            data-currentday=${content.currentDay}
-            >
-            ${content.date}
-        </button>
-    </div>
-    `
-    return element
+export function findElemByDate( date ){
+
+    const buttons = Array.from( document.querySelectorAll("#month button") )
```

Comment:
considering you already gave data attribute on the element, you can just select by it